lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1749,6 +1754,36 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
     if (BS->SPIRVMemoryAccess::isNonTemporal())
       transNonTemporalMetadata(SI);
     transAliasingMemAccess<SPIRVStore>(BS, SI);
+
+    // In SPIRV Writer pass, the SPIRV Builtin variables were processed
+    // in a unique way to ensure we adhered to OpenCL SPIRV Env Spec.
+    // The global variable was copied into a function-level storage.
+    // The following block of code identifies those changes and reverts them.
+    auto *StorePtr = SI->getPointerOperand();
+    if (StorePtr->hasName() &&
+        StorePtr->getName().starts_with("__local_SPIRV_Builtin")) {
+      if (auto *LI = dyn_cast<LoadInst>(SI->getValueOperand())) {
```

Comment:
Are these if's necessary?  Is it possible that some __local_SPIRV_Builtin's will not be removed in reverse translation?

Reply:
Good point. They can be turned to 'cast' statements and then asserts to catch any unexpected behavior.

Thanks